app/uk/gov/hmrc/gform/controllers/AuthenticatedRequestActions.scala

```diff
@@ -121,10 +121,16 @@ class AuthenticatedRequestActions(
     f: Request[AnyContent] => LangADT => AuthCacheWithoutForm => Future[Result]): Action[AnyContent] = Action.async {
     implicit request =>
       implicit val l: LangADT = getCurrentLanguage(request)
+
       for {
         formTemplate <- gformConnector.getFormTemplate(formTemplateId)
         authResult <- authService
-                       .authenticateAndAuthorise(formTemplate, request.uri, getAffinityGroup, ggAuthorised(request))
+                       .authenticateAndAuthorise(
+                         formTemplate,
+                         request.uri,
+                         getAffinityGroup,
+                         ggAuthorised(request),
+                         request.cookies.get("caseworker-assumed-identity"))
```

Comment:
Literal string `"caseworker-assumed-identity"` is mention twice. This will result in a bug once name of that cookie will change in the future and will be subsequently updated only on one place.

Reply:
Changed it to read the cookie-name from the configuration file